bitarray/_bitarray.c

```diff
@@ -443,19 +548,59 @@ static int bitcount_lookup[256] = {
 
 /* returns number of 1 bits */
 static idx_t
-count(bitarrayobject *self)
+count_range(bitarrayobject *self, Py_ssize_t start, Py_ssize_t end)
 {
     Py_ssize_t i;
     idx_t res = 0;
     unsigned char c;
 
     setunused(self);
-    for (i = 0; i < Py_SIZE(self); i++) {
+    for (i = start; i < end; i++) {
         c = self->ob_item[i];
         res += bitcount_lookup[c];
     }
     return res;
 }
+static idx_t
+count_old(bitarrayobject *self)
+{
+	setunused(self);
+	return self->nbits_set = count_range(self, 0, Py_SIZE(self));
+}
+
+static idx_t
+count_alt(bitarrayobject *self)
```

Comment:
There are several `*_alt` and `*_old` methods in the code, which are also exposed in the API.  Could you clarify what those are?  Left for benchmarking purposes?